File: uniffi_bindgen/src/bindings/kotlin/templates/Int64Helper.kt
internal object FfiConverterI64 {
    fun lift(v: Long): Long {
        return v
    }

    fun read(buf: ByteBuffer): Long {
        return buf.getLong()
    }

    fun lower(v: Long): Long {
        return v
    }

    fun write(v: Long, buf: RustBufferBuilder) {
        buf.putLong(v)
    }
}

File: uniffi_bindgen/src/bindings/swift/templates/Int64Helper.swift
fileprivate struct FfiConverterInt64: FfiConverterPrimitive {
    typealias FfiType = Int64
    typealias SwiftType = Int64

    static func read(from buf: inout (data: Data, offset: Data.Index)) throws -> Int64 {
        return try lift(readInt(&buf))
    }

    static func write(_ value: Int64, into buf: inout [UInt8]) {
        writeInt(&buf, lower(value))
    }
}
